DA-644: Short-Term Memory with Crew-AI and Couchbase

teetangh opened the pull request.

deniswsrosa commented:
@teetangh I would like to congratulate you on the 2 months anniversary of your PR

teetangh commented:
Hey @deniswsrosa,
Apologies for the delay in getting this PR merged. I'm still waiting for the final review and feedback from @nithishr. I'm following up to get it reviewed ASAP and will address any required changes quickly. Hoping to get it merged soon.

nithishr commented:
@teetangh Few things to improve:
* Please explain what memory is & how it is useful in an Agent.
* Typo in `Load Sensitive Information`
* The example mentions that Couchbase does not have Vector Search, which we should not publish in the tutorial.
* Are you sure that the crew ai agent is using Couchbase as the short-term memory? There are references to ChromaDB in the logs. I suspect that ChromaDB is the default & being used for the memory. Only the explicit memory reference is being powered by Couchbase.
* [Minor] You don't need to install couchbase SDK. It will get installed with langchain-couchbase.
* Conclusion/summary is missing

teetangh commented:
@nithishr Done:

- Explained what memory is and how it helps an Agent.
- Fixed the typo in "Load Sensitive Information."
- Removed the statement about Couchbase not having Vector Search.
- Verified memory usage in Crew AI—ChromaDB was the default memory storage in the crew class. Now changed it to use the CouchbaseStorage class.
- Removed the separate Couchbase SDK installation step since it's included with langchain-couchbase.
- Added a conclusion/summary section.

nithishr commented:
@teetangh Looks much better. A couple of suggestions:
* Can we show the information that is stored in Couchbase when the Agent is running? Could be a single SQL++ query or even a simple screenshot of the data that is being stored in Couchbase with an explanation. It is still not clear to me without running the code about the information being stored in the memory when the agent is running.
* Also, I believe the Couchbase Storage that you have written can be used for long term storage as well. If so, please add a note on the same. 
* I think the memory section is coming from the CrewAI docs. If so, please add a reference to it.

teetangh commented:
@nithishr added both notes and attached screenshot as you mentioned

nithishr commented:
@teetangh Are you sure that something is getting stored in Couchbase during the execution of the Crew AI agent? The response I see in the memory is the one that you store earlier. You are getting two results but only one from before is returned with a score threshold of 0 which I interpret as nothing got stored in the Memory during the agent run. 
Can you double check?

Also, with the note on CrewAI documentation, please add the link to their docs.

teetangh commented:
@nithishr added both notes

nithishr commented on crewai-short-term-memory/CouchbaseStorage_Demo.ipynb:
Can you change the SDK link to https://docs.couchbase.com/python-sdk/current/howtos/full-text-searching-with-sdk.html#single-vector-query

teetangh replied:
added

nithishr commented on crewai-short-term-memory/README.md:
Please add the notes here as well.

teetangh replied:
added

nithishr reviewed (dismissed):
Please add the open items & merge the PR
